velociraptor/autoplotter/plot.py

```diff
@@ -87,6 +87,32 @@ def mass_function(
     return fig, ax
 
 
+def luminosity_function(
+    x: unyt.unyt_array, x_bins: unyt.unyt_array, luminosity_function: VelociraptorLine
+) -> Tuple[plt.Figure, plt.Axes]:
+    """
+    Creates a plot of x as a luminosity function, binned with x_bins.
+    """
+
+    fig, ax = plt.subplots()
+
+    centers, luminosity_function, error, *_ = luminosity_function.output
+
+    # A Bit of a dirty hack but we want add a formally-dimensionless magnitude unit to the y-axis
+    unyt.unit_registry.default_unit_registry.add(
+        "mag", 1.0, dimensions=unyt.dimensions.dimensionless
+    )
+    luminosity_function = unyt.unyt_array(
+        luminosity_function.v,
+        str(luminosity_function.units) + " * mag**-1",
```

Comment:
```suggestion
        luminosity_function.units / unyt.mag,
```

Reply:
Will that work if mag is not a default unyt unit?

Reply:
Ah I see what you've done here, sorry. I didn't read it correctly.

You should add your unit in `https://github.com/SWIFTSIM/velociraptor-python/blob/master/velociraptor/__init__.py`. 
```
In [1]: import unyt

In [4]: unyt.define_unit("example_unit", 1.0 * unyt.kg, tex_repr=r"E")

In [5]: unyt.example_unit
Out[5]: example_unit
```

Whilst you're there, can you add a 1.0 * (base unit) to the ones for msun and hydrogen mass?

Reply:
That's now done. And works correctly. Not sure why my original version did not like just using `define_unit()`